source/adl-standalone/tests/connection.yml

```diff
@@ -0,0 +1,42 @@
+tests:
```

Comment:
This test format is not described in the README.  

I'm also not sure what the point of it is, as they look to be just unit tests.

Reply:
If the intention is that they are integration tests, we would have to say what to do in the integration test.   Presumably it would involve running at least one command for each, and either getting an exception or not depending on whether it's valid or not.

Reply:
I don't think these tests add much value. To relate this back to the [scope](https://docs.google.com/document/d/1eXwy-7v_7BlPpcr7emXWSJZWm4V47ZuH5lvBj-XnOmY/edit#), we had the following requirements:

> * Drivers should be able to connect to ADL. ADL pretends to be a standalone MongoDB server.
> * Drivers should be able to authenticate using both SCRAM-SHA-1 and SCRAM-SHA-256.

I don't think the first bullet needs its own test, as the CRUD tests in this PR adequately cover that.

I would propose adding two prose tests for drivers to manually configure the appropriate authentication for mongohouse and then connect to it and issue a single command to verify that authentication was successful (e.g. `ping` if that's allowed).

https://github.com/10gen/mongohouse/blob/master/README.md#getting-started states that "All the sample config files have authentication enabled", but I didn't find reference to auth mechanisms in the repo apart from some files in the [`internal/` directory](https://github.com/10gen/mongohouse/search?q=SCRAM-SHA-256&unscoped_q=SCRAM-SHA-256). @craiggwilson: Can you chime in here? Are there documented instructions for configuring mongohouse with SCRAM-SHA-1 and SCRAM-SHA-256?

Reply:
The auth for the sample/used configs all have auth just built-in. Nothing needs to happen other than to use one of the config files.

Reply:
Ah, I missed the link to https://github.com/10gen/mongohouse/tree/master/testdata/config in https://github.com/10gen/mongohouse/blob/master/README.md#configuration.

So, to confirm, if our tests use [single-local-local.yaml](https://github.com/10gen/mongohouse/blob/master/testdata/config/single-local-local.yaml) for everything, we can trust that this has _both_ SCRAM-SHA-1 _and_ SCRAM-SHA-256 available for `mhuser:pencil` based on https://github.com/10gen/mongohouse/blob/4e6c3c13b085545e573ba2e41e66ae7de03c8211/testdata/config/single-local-local.yaml#L35?

@jstewart-mongo: I see that your "To build ``mongohoused`` for local testing" section already loads this configuration, so if you do get rid of that section (per my earlier comment about build instructions being redundant if we provision through drivers-evergreen-tools scripts), I think you'll still want to leave in the bit about how to launch `mongohoused` with the appropriate configuration file.

Reply:
Yes, if you use `single-local-local.yaml`, then it has that user attached. 

That being said, I'd highly suggest creating your own config and referencing your own "data" files. We on ADL may change this config file and or the data files it points to for our own tests every once in a while and that will definitely break this. Should be easy enough to copy `single-local-local` and some of the tests into a new place and just use those.

Reply:
I concur, it seems most reasonable to have drivers-evergreen-tools contain its own ADL configuration _and_ a script to provision ADL from your S3 bucket.